contests/_27-PDP/a-xxx-solution.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: solution
+codename: xxx
+---
+
+## Brute force λύση
+
+Πρέπει να βρούμε πόσα παιδιά βλέπουν την είσοδο του κυλικείου. Γνωρίζουμε ότι ένα παιδί βλέπει την είσοδο αν και μόνο αν όλα τα παιδιά που στέκονται μπροστά του έχουν ύψος μικρότερο από το δικό του.
+
+Επομένως, μία σκέψη θα ήταν για κάθε παιδί να ελέγχουμε αν είναι ψηλότερο από όλα τα παιδιά που βρίσκονται μπροστά του. Δηλαδή για κάθε παιδί, $$i$$, να διατρέχουμε τον πίνακα $$A$$, στον οποίο έχουμε αποθηκεύσει τα ύψη των παιδιών, από τη θέση $$j = i + 1$$ μέχρι και την $$j = N - 1$$ και να βλέπουμε αν όλα τα $$A[j]$$ είναι μικρότερα από το $$A[i]$$.
+
+Η boolean μεταβλητή $$ok$$ που χρησιμοποιούμε στον κώδικα που φαίνεται παρακάτω αρχικοποιείται σε $$true$$ για κάθε παιδί $$i$$ και γίνεται $$false$$ αν βρεθεί κάποιο $$j > i: A[j] \geq A[i]$$, δηλαδή αν υπάρχει κάποιο παιδί μπροστά από το $$i$$-οστό το οποίο έχει ύψος μεγαλύτερο ή ίσο από το δικό του, εμποδίζοντάς το να δει την είσοδο.
+
+Το πόσα παιδιά βλέπουν το κυλικείο θα το μετράμε με τη βοήθεια της μεταβλητής $$K$$. Η μεταβλητή $$K$$ θα είναι αρχικοποιημένη στο $$0$$ και όποτε συναντάμε ένα παιδί που μπορεί να δει το κυλικείο, δηλαδή η μεταβλητή $$ok$$ παραμένει $$true$$ για αυτό, θα την αυξάνουμε κατά ένα.
+
+Όταν ελέγξουμε όλα τα παιδιά, δηλαδή όλα τα $$i \in \{0, 1, \dots, N - 1\}$$ η μεταβλητή $$K$$ θα αποτελεί τη ζητούμενη απάντηση, οπότε αρκεί να την εκτυπώσουμε.
+
+Η πολυπλοκότητα του αλγορίθμου είναι $$O(N^2)$$ και χρειάζεται $$O(N)$$ μνήμη. Η πολυπλοκότητα αυτή δεν είναι αρκετά καλή και παρόλο που θα επιτρέψει στη λύση μας να πάρει κάποιες μονάδες, δε θα μπορέσει να περάσει όλα τα testcases λόγω υπέρβασης του χρονικού ορίου εκτέλεσης.
+
+{% highlight c++ %}
+#include <cstdio>
+using namespace std;
+
+const int MAXN = 1e6 + 5;
+
+int A[MAXN];
+
+int main() {
+	freopen("xxx.in", "r", stdin);
+	freopen("xxx.out", "w", stdout);
+	int N;
```

Comment:
To ίδιο και εδώ και παρακάτω.

Reply:
Ας κάνουμε merge τώρα και τα διορθώνουμε σε μετέπειτα commits.